platform/android/export/export.cpp

```diff
@@ -2274,109 +2347,256 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
+		String path = static_cast<String>(p_preset->get(launcher_icon_option)).strip_edges();
+		if (path.empty() || ImageLoader::load_image(path, launcher_icon_image) != OK) {
+			ImageLoader::load_image(project_icon_path, launcher_icon_image);
+		}
 
-			ERR_FAIL_COND_V_MSG(sdk_path == "", ERR_UNCONFIGURED, "Android SDK path must be configured in Editor Settings at 'export/android/custom_build_sdk_path'.");
+		// Adaptive foreground: user selection -> regular icon (user selection -> project icon -> default).
+		path = static_cast<String>(p_preset->get(launcher_adaptive_icon_foreground_option)).strip_edges();
+		if (path.empty() || ImageLoader::load_image(path, launcher_adaptive_icon_foreground_image) != OK) {
+			launcher_adaptive_icon_foreground_image = launcher_icon_image;
+		}
 
-			_update_custom_build_project();
+		// Adaptive background: user selection -> default.
+		path = static_cast<String>(p_preset->get(launcher_adaptive_icon_background_option)).strip_edges();
+		if (!path.empty()) {
+			ImageLoader::load_image(path, launcher_adaptive_icon_background_image);
+		}
 
-			OS::get_singleton()->set_environment("ANDROID_HOME", sdk_path); //set and overwrite if required
+		//        String file = "res://android/build/AndroidManifest.xml";
+		//
+		//        Vector<uint8_t> android_manifest_file_data = FileAccess::get_file_as_array(file);
+		//        _fix_manifest(p_preset, android_manifest_file_data,  p_flags & (DEBUG_FLAG_DUMB_CLIENT | DEBUG_FLAG_REMOTE_DEBUG));
+
+		for (int i = 0; i < icon_densities_count; ++i) {
+			//void _process_launcher_icons(const String &p_processing_file_name, const Ref<Image> &p_source_image, const LauncherIcon p_icon, Vector<uint8_t> &p_data)
+			if (launcher_icon_image.is_valid() && !launcher_icon_image->empty()) {
+				Vector<uint8_t> data = resize_launcher_icon(launcher_icon_image, launcher_icons[i]);
+				String img_path = launcher_icons[i].export_path;
+				img_path = img_path.insert(0, "res://android/build/");
+				store_in_gradle_project(img_path, data, Z_NO_COMPRESSION);
+			}
+			if (launcher_adaptive_icon_foreground_image.is_valid() && !launcher_adaptive_icon_foreground_image->empty()) {
+				Vector<uint8_t> data = resize_launcher_icon(launcher_adaptive_icon_foreground_image, launcher_adaptive_icon_foregrounds[i]);
+				String img_path = launcher_adaptive_icon_foregrounds[i].export_path;
+				img_path = img_path.insert(0, "res://android/build/");
+				store_in_gradle_project(img_path, data, Z_NO_COMPRESSION);
+			}
+			if (launcher_adaptive_icon_background_image.is_valid() && !launcher_adaptive_icon_background_image->empty()) {
+				Vector<uint8_t> data = resize_launcher_icon(launcher_adaptive_icon_background_image, launcher_adaptive_icon_backgrounds[i]);
+				String img_path = launcher_adaptive_icon_backgrounds[i].export_path;
+				img_path = img_path.insert(0, "res://android/build/");
+				store_in_gradle_project(img_path, data, Z_NO_COMPRESSION);
+			}
+		}
+
```

Comment:
Add a `TODO: reimplement these features on the raw project rather than as an APK`